### PR TITLE
fix: missing release notes

### LINKED ISSUE #3092: Missing release notes for django-filter, github release notes available
**What Renovate type are you using?**
Renovate GitHub App

**Describe the bug**
release notes in github releases (7h ago): https://github.com/carltongibson/django-filter/releases
package to upgrade: django-filter
PR created 6h ago (release notes were therefore already available) without release notes:

### PR DESCRIPTION
**Pull request by nihalwasheretoo**
This hotfix fixes the missing release notes issue for pypi packages. The problem was with source-url being set to null even if it existed. This was happening because it was not being parsed correctly. 

Closes #3092

### CODE REVIEW

**Comment by rarkins** [comment]
Can you add a test?

**Comment by nihal9ns** [comment]
> Can you add a test?

sure

File: lib/datasource/metadata.js [hunk 1]
```diff
@@ -100,4 +105,6 @@ function addMetaData(dep, datasource, lookupName) {
       delete dep[url];
     }
   }
+  /* eslint-disable consistent-return */
```
Comment (rarkins) [inline lib/datasource/metadata.js]: Why is this needed?

**Comment by rarkins** [comment]
Is this because some of our manual source URLs are wrong though? What's an example one that's currently wrong?

File: lib/datasource/metadata.js (hunk 1 above)
Comment (nihal9ns) [reply lib/datasource/metadata.js]: It was needed because I added a return statement there which was not there earlier, 
`(return dep;)` . Lint started throwing a no return statement error for that function. I don't know why that function should not have a return statement?

File: lib/datasource/metadata.js [hunk 2]
```diff
@@ -79,7 +79,12 @@ function addMetaData(dep, datasource, lookupName) {
     dep.sourceUrl = manualSourceUrls[datasource][depName];
   }
   if (dep.sourceUrl && dep.sourceUrl.includes('github.com')) {
-    dep.sourceUrl = parse(dep.sourceUrl);
+    dep.sourceUrl = parse(
```
Comment (nihal9ns) [inline lib/datasource/metadata.js]: https://github.com/renovatebot/renovate/pull/4123#issuecomment-513137483

Our source url's are not wrong. It's just that we needed to add a massage for source urls so that they can be parsed here correctly, a source url like `https://github.com/carltongibson/django-filter/tree/master` was returning undefined after being parsed.

File: lib/datasource/metadata.js (hunk 1 above)
Comment (nihal9ns) [reply lib/datasource/metadata.js]: I needed that return statement so that I could get the data back from the addMetaData() in the test. It doesn't break anything.

**Comment by rarkins** [comment]
I think it should be possible to modify the passed object like is done in the app (instead of returning a value)

**Comment by nihal9ns** [comment]
> object

Let me check

**Comment by nihal9ns** [comment]
Let me check

**Comment by nihal9ns** [comment]
Can you lead me to an example in the app where we are testing a function which does not return anything?

**Comment by nihal9ns** [comment]
> I think it should be possible to modify the passed object like is done in the app (instead of returning a value)

I think you got confused with what I said. 

I added the return statement only so that I could simply require the function `addMetaData()` in my test and call the function and the function would return me the `dep` object and then I can expect and apply assertions on that result of that function.

Makes sense?

**Comment by rarkins** [comment]
You can pass dep and then check it’s fields after calling the function. It does not need to be returned.

**Comment by nihal9ns** [comment]
Let me try

**Comment by rarkins** [comment]
@nihal9ns this PR also has the same CI problem. Can you make sure your fork's master is in sync with the main repo and then recreate this PR?
